xtokens/src/lib.rs

```diff
@@ -110,18 +130,18 @@ pub mod module {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Transfer native currencies.
-		#[transactional]
-		#[pallet::weight(1000)]
+		#[pallet::weight(Pallet::<T>::weight_of_transfer(currency_id.clone(),*amount, &dest))]
```

Comment:
```suggestion
		#[pallet::weight(Pallet::<T>::weight_of_transfer(*currency_id, *amount, &dest))]
```

Reply:
`currency_id` can't be moved here. I'll update the format.